File: library-ui/src/app/models/book.ts
export class Book {
    public id: number;
    public category: string;
    public issueDate: Date;
    public title: string;
    public authorId: number;

    constructor(id: number, category: string, issueDate: Date, title: string, authorId: number) {
        this.id = id;
        this.category = category;
        this.issueDate = issueDate;
        this.title = title;
        this.authorId = authorId;
    }
}

File: src/main/java/com/kris/library/entities/Book.java
package com.kris.library.entities;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.util.Date;

/**
 * @author kristina.drashkova
 */
@Entity
@Table(name = "books")
@Getter
@Setter
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String title;

    private Category category;

    @ManyToOne
    private Author author;

    @Column(name = "issue_date")
    private Date issueDate;
}
